src/middlewared/middlewared/plugins/post_install.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+
+import jsonschema
+
+from middlewared.service import Service
+
+PATH = "/data/post_install.json"
+
+SCHEMA = {
+    "type": "object",
+    "additionalProperties": False,
+    "properties": {
+        "network_interface": {
```

Comment:
We can have this as a required property to make sure we don't have an empty dict ?

Reply:
Also maybe this should be `network_interfaces` ?

Reply:
What's the problem with having an empty dict, why does network interface configuration have to be required?

Reply:
There shouldn't be any, idea was consistency and being strict about values where this can be an empty list but at least the key is there so we don't have to always be safe about it when retrieving values.